apis/python/src/tiledb/vector_search/embeddings/colpali_embedding.py

```diff
@@ -0,0 +1,102 @@
+from typing import Dict, OrderedDict, Tuple
+
+import numpy as np
+
+from tiledb.vector_search.embeddings import ObjectEmbedding
+
+EMBED_DIM = 128
+
+
+class ColpaliEmbedding(ObjectEmbedding):
+    def __init__(
+        self,
+        model_name: str = "vidore/colpali-v1.2",
+        device: str = None,
+        batch_size: int = 4,
+    ):
+        self.model_name = model_name
+        self.device = device
+        self.batch_size = batch_size
+        self.model = None
+        self.processor = None
+
+    def init_kwargs(self) -> Dict:
+        return {
+            "model_name": self.model_name,
+            "device": self.device,
+            "batch_size": self.batch_size,
+        }
+
+    def dimensions(self) -> int:
+        return EMBED_DIM
+
+    def vector_type(self) -> np.dtype:
+        return np.float32
+
+    def load(self) -> None:
+        import torch
+        from colpali_engine.models import ColPali
+        from colpali_engine.models import ColPaliProcessor
+
+        if self.device is None:
+            if torch.cuda.is_available() and torch.cuda.device_count() > 0:
+                self.device = "cuda"
+            else:
+                self.device = "cpu"
```

Comment:
Maybe we could add `mps` as a third option? Likely not a super common use case, but I always appreciate if repos have this when I run pytorch things locally. And they do call it out here: https://github.com/illuin-tech/colpali/blob/7f53f53ed005cfbafd2640ca7705a6297a97971f/README.md?plain=1#L69

Reply:
Added